### ISSUE TITLE
Connecting to a ucx cluster created from command line fails with latest dask/distributed

### ISSUE THREAD

**Issue by ayushdg** [rapidsai/dask-cuda#772]
Steps to reproduce:
Start scheduler:
```bash
DASK_DISTRIBUTED_UCX__CUDA_COPY=True DASK_DISTRIBUTED_UCX__TCP=True DASK_DISTRIBUTED_UCX__NVLINK=True DASK_DISTRIBUTED_UCX__INFINIBAND=False DASK_DISTRIBUTED_UCX__RDMACM=False nohup dask-scheduler \
  --port 8786 \
  --protocol ucx &
```
Start worker:
```bash
nohup dask-cuda-worker \
  --enable-tcp-over-ucx \
  --enable-nvlink \
  --disable-infiniband \
  --disable-rdmacm ucx://ip-addr:8786 &
```
Connect to cluster:
```python
from distributed import Client
c = Client("ucx://ip-addr:8786") # hangs and eventually fails with timeout
```

This fails with dask/distributed `2021.11.0` but succeeds with `2021.10.0` 
dask-cuda nightly version `21.12.00a211105`

**Comment by pentschev** [comment]
It's not clear whether you're setting variables for the client in any way, but you should always set the same variables to run the client as you do for the scheduler, see https://docs.rapids.ai/api/dask-cuda/nightly/examples/ucx.html#client . The example uses `initialize`, but setting the environment variables should work the same.

I can't reproduce this locally though, even if I don't set the environment variables when running the client, and these are the Dask packages I was running with:

```
$ conda list dask
# packages in environment at /datasets/pentschev/miniconda3/envs/rn-112-21.12.211106:
#
# Name                    Version                   Build  Channel
dask                      2021.11.0          pyhd8ed1ab_0    conda-forge
dask-core                 2021.11.0          pyhd8ed1ab_0    conda-forge
dask-cuda                 21.12.00a211106         py38_28    rapidsai-nightly
dask-cudf                 21.12.00a211105 cuda_11.2_py38_g4cba672c17_224    rapidsai-nightly
$ conda list distributed
# packages in environment at /datasets/pentschev/miniconda3/envs/rn-112-21.12.211106:
#
# Name                    Version                   Build  Channel
distributed               2021.11.0        py38h578d9bd_0    conda-forge
```

**Comment by ayushdg** [comment]
Thanks for the response. I did some followup testing and can confirm two things:
1. Setting the client environment with `initialize` allowed me to connect to the cluster, while probably incorrect,  on distributed `2021.10.0` and prior things worked even without the `initialize` step.
2. I was also able to confirm that adding the revert patch https://github.com/dask/distributed/pull/5505 also fixed the issue where I was able to connect without the `initialize` step. 

Moving forward I'll add `initialize` to my scripts, but it might be worth exploring why connecting to the cluster (without init) broke with those changes.
